esphome/components/adc/adc_sensor_common.cpp

```diff
@@ -6,6 +6,60 @@ namespace adc {
 
 static const char *const TAG = "adc.common";
 
+const char *sampling_mode_str(SamplingMode mode) {
+  switch (mode) {
+    case SamplingMode::AVG:
+      return "average";
+    case SamplingMode::MIN:
+      return "minimum";
+    case SamplingMode::MAX:
+      return "maximum";
+  }
+
+  return "unknown";
+}
```

Comment:
```suggestion
const LogString *sampling_mode_to_str(SamplingMode mode) {
  switch (mode) {
    case SamplingMode::AVG:
      return LOG_STR("average");
    case SamplingMode::MIN:
      return LOG_STR("minimum");
    case SamplingMode::MAX:
      return LOG_STR("maximum");
  }

  return LOG_STR("unknown");
}
```
https://github.com/esphome/esphome/blob/fc847c1de8927871e783db1a35e967efabda9f04/esphome/core/log.h#L166